aten/src/ATen/native/cuda/jit_utils.cpp

```diff
@@ -932,6 +971,13 @@ std::string generate_reduction_code(
         env.s("complex_math_string", "");
         env.s("complex", std::to_string(0));
       }
+      if (f_inputs_type == "std::complex<at::Half>" ||
+          result_type == "std::complex<at::Half>") {
+        TORCH_CHECK(
+            false, "complex<Half> reduction not supported by JITERATOR");
```

Comment:
Error if reduction gets input or output of complex<Half>

Reply:
Why is complex half reduction not supported? It should not be different from regular low precision reductions that are supported.

Reply:
I was planning to add it in a different PR and port an operator with test.

Reply:
Cool, minor changes will be required to add complex half string, right? warp shuffles should work as is, as they should operate on accum type.

Reply:
I am assuming that it would be a smaller change. (I haven't had a look though).